k8s/tools/cleanup-pubsub-gke.sh

```diff
@@ -0,0 +1,146 @@
+read response
+if [[ $response == "delete all" ]] ; then
+  echo "'delete all' specified. Exiting."
+  exit 1
+fi
+
+# Check configured gcloud project
+if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+  DEVSHELL_PROJECT_ID=$(gcloud config get-value project)
+  ERRMSG="ERROR: Could not get configured project. Please either restart "
+  ERRMSG+="Google Cloudshell, or set configured project with "
+  ERRMSG+="'gcloud config set project PROJECT' when running outside of Cloudshell."
+  if [[ -z "$DEVSHELL_PROJECT_ID" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+  echo "Environment variable \$DEVSHELL_PROJECT_ID was not set at start time "
+  echo "so attempting to get project config from gcloud config."
+  echo -n "Do you want to use $DEVSHELL_PROJECT_ID as the target project? (y / n) > "
+  read response
+  if [[ $response != "y" && $response != "Y" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+fi
+
+# Delete the cluster
+if [[ "$*" != *--no-cluster* ]] ; then
+  echo "Deleting cluster $CLUSTER_NAME"
+  gcloud -q --project $DEVSHELL_PROJECT_ID container clusters delete $CLUSTER_NAME --zone $ZONE
+fi
+
+# Delete the GCS storage bucket
+if [[ "$*" != *--no-gcs* ]] ; then
+  echo "Deleting GCS storage bucket gs://$INSTANCE_ID"
+  gsutil -q rm -r gs://$INSTANCE_ID
+fi
+
+# Delete PubSub topics
+if [[ "$*" != *--no-pubsub* ]] ; then
+  echo "Deleting PubSub topic $INSTANCE_ID"
+  gcloud -q --project $DEVSHELL_PROJECT_ID pubsub topics delete $INSTANCE_ID
+  gcloud -q --project $DEVSHELL_PROJECT_ID pubsub topics delete "$INSTANCE_ID-psq"
+
+  # Delete PubSub subscriptions
+  gcloud -q --project $DEVSHELL_PROJECT_ID pubsub subscriptions delete $INSTANCE_ID
+  gcloud -q --project $DEVSHELL_PROJECT_ID pubsub subscriptions delete "$INSTANCE_ID-psq"
+fi
+
+# Delete the Filestore instance
+if [[ "$*" != *--no-filestore* ]] ; then
+  echo "Deleting Filestore instance $FILESTORE_NAME"
+  gcloud -q --project $DEVSHELL_PROJECT_ID filestore instances delete $FILESTORE_NAME --zone $ZONE
+fi
+
+# Remove cloud functions
+if [[ "$*" != *--no-cloudfunctions* ]] ; then
+  echo "Delete Google Cloud functions"
+  if gcloud functions --project $DEVSHELL_PROJECT_ID list | grep gettasks; then
+    gcloud --project $DEVSHELL_PROJECT_ID -q functions delete gettasks --region $REGION
+  fi
+  if gcloud functions --project $DEVSHELL_PROJECT_ID list | grep closetask; then
+    gcloud --project $DEVSHELL_PROJECT_ID -q functions delete closetask --region $REGION
+  fi
+  if gcloud functions --project $DEVSHELL_PROJECT_ID list | grep closetasks; then
+    gcloud --project $DEVSHELL_PROJECT_ID -q functions delete closetasks  --region $REGION
+  fi
+fi
+
+# Cleanup Datastore indexes
+if [[ "$*" != *--no-datastore* ]] ; then
+  echo "Cleaning up Datastore indexes"
+  gcloud --project $DEVSHELL_PROJECT_ID -q datastore indexes cleanup ../tools/gcf_init/index.yaml
+fi
+
+# Use local `gcloud auth` credentials rather than creating new Service Account.
```

Comment:
Wait, doesn't all this auth stuff need to happen at the start of the script?

Reply:
Yeah you're right for auth reasons, moved to top